src/Cardano/Address/Style/Jormungandr.hs

```diff
@@ -0,0 +1,314 @@
+            coinTypeXPrv = -- lvl2 derivation; hardened derivation of coin_type'
+                deriveXPrv DerivationScheme2 purposeXPrv coinTypeIx
+            acctXPrv = -- lvl3 derivation; hardened derivation of account' index
+                deriveXPrv DerivationScheme2 coinTypeXPrv accIx
+        in
+            Jormungandr acctXPrv
+
+    deriveAddressPrivateKey (Jormungandr accXPrv) accountingStyle addrIx =
+        let
+            changeCode =
+                toEnum @(Index 'Soft _) $ fromEnum accountingStyle
+            changeXPrv = -- lvl4 derivation; soft derivation of change chain
+                deriveXPrv DerivationScheme2 accXPrv changeCode
+            addrXPrv = -- lvl5 derivation; soft derivation of address index
+                deriveXPrv DerivationScheme2 changeXPrv addrIx
+        in
+            Jormungandr addrXPrv
+
+instance SoftDerivation Jormungandr where
+    deriveAddressPublicKey (Jormungandr accXPub) accountingStyle addrIx =
+        fromMaybe errWrongIndex $ do
+            let changeCode = toEnum @(Index 'Soft _) $ fromEnum accountingStyle
+            changeXPub <- -- lvl4 derivation in bip44 is derivation of change chain
+                deriveXPub DerivationScheme2 accXPub changeCode
+            addrXPub <- -- lvl5 derivation in bip44 is derivation of address chain
+                deriveXPub DerivationScheme2 changeXPub addrIx
+            return $ Jormungandr addrXPub
+      where
+        errWrongIndex = error $
+            "deriveAddressPublicKey failed: was given an hardened (or too big) \
+            \index for soft path derivation ( " ++ show addrIx ++ "). This is \
+            \either a programmer error, or, we may have reached the maximum \
+            \number of addresses for a given wallet."
+
+instance HasNetworkDiscriminant Jormungandr where
+    type NetworkDiscriminant Jormungandr = (AddressDiscrimination, NetworkTag)
+    addressDiscrimination = fst
+    networkTag = snd
+
+instance PaymentAddress Jormungandr where
+    paymentAddress discrimination k = unsafeMkAddress $
+        invariantSize expectedLength $ BL.toStrict $ runPut $ do
+            putWord8 firstByte
+            putByteString (getPublicKey $ getKey k)
+      where
+          firstByte = case addressDiscrimination @Jormungandr discrimination of
+              RequiresNetworkTag -> 0x83
+              RequiresNoTag -> 0x03
+          expectedLength = 1 + publicKeySize
+
+instance DelegationAddress Jormungandr where
+    delegationAddress discrimination paymentKey stakingKey = unsafeMkAddress $
+        invariantSize expectedLength $ BL.toStrict $ runPut $ do
+            putWord8 firstByte
+            putByteString . getPublicKey . getKey $ paymentKey
+            putByteString . getPublicKey . getKey $ stakingKey
+      where
+          firstByte = case addressDiscrimination @Jormungandr discrimination of
+              RequiresNetworkTag -> 0x84
+              RequiresNoTag -> 0x04
```

Comment:
These constants should be the value behind `networkTag @Jormungandr`. So here, I'd suggest to simply export a single smart constructor as:

```hs
incentivizedTestnet :: NetworkDiscriminat Jormungandr
incentivizedTestnet = (RequiresNetworkTag, 0x84)
``` 

And, simply use `networkTag @Jormungandr` for the first byte. The purpose of the `HasNetworkDiscriminant` abstraction is to capture these constants and offer a consistant API to the caller.

Reply:
done